### PR TITLE
Use stream/group name in CF appname.

### PR DESCRIPTION
**Pull request by ericbottard**
This fixes spring-cloud/spring-cloud-dataflow-admin-cloudfoundry#75 
This also fixes spring-cloud/spring-cloud-dataflow-admin-cloudfoundry#74

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java [hunk 1]
```diff
@@ -158,12 +160,12 @@ public AppStatus status(String id) {
...
 
-	/**
-	 * TODO: Should we join properties with override, or replace?
-	 *
-	 * @param request
-	 * @return
-	 */
+	private String deploymentId(AppDeploymentRequest request) {
+		return String.format("%s-%s",
+				request.getEnvironmentProperties().get(GROUP_PROPERTY_KEY),
```
Comment (markpollack) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: Where does this come from?

Comment (ericbottard) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: Comes from the caller, as a well known AppDeployer property. See [here](https://github.com/spring-cloud/spring-cloud-deployer/blob/6e44c8ed742a890cf2131c79ca61bb6af74a7513/spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/AppDeployer.java#L39-L42). Not very clear though if it is mandatory. cc @markfisher

Comment (markfisher) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: it should be implicitly required for a stream app (but not for a task app) - could possibly throw `IllegalArgumentException` *here* if it's not provided

Comment (ericbottard) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: Then AppDefinition accepting a `null` properties map should be changed, should it not?

Comment (markpollack) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: That sounds reasonable, I'll create an issue for that when this is closed and also add the `IllegalArgumentException` that Mark mentioned.

**Comment by markpollack** [comment]
verified
```
dataflow:>stream create --name tickytocky --definition "time | log" --deploy 
Created and deployed new stream 'tickytocky'
dataflow:>! cf apps
command is:cf apps
Getting apps in org mpollack-org / space development as mpollack...
OK

name                 requested state   instances   memory   disk   urls   
mlp-dataflow-admin   started           1/1         1G       1G     mlp-dataflow-admin.navy.springapps.io   
tickytocky-log       started           0/1         1G       1G     tickytocky-log.navy.springapps.io   
tickytocky-time      started           0/1         1G       1G     tickytocky-time.navy.springapps.io   
```

**Comment by markpollack** [comment]
Create https://github.com/spring-cloud/spring-cloud-deployer/issues/47

**Comment by markpollack** [comment]
Addressed comments and closed as 9f32ebb7a161df1c1fd0a7e8c7e4a1678898555b

**Pull request closed without merging**